Modules/Microsoft365DSC/DSCResources/MSFT_EXOMigrationConfig/MSFT_EXOMigrationConfig.psm1

```diff
@@ -0,0 +1,453 @@
+            CertificatePassword                        = $CertificatePassword
+            Managedidentity                            = $ManagedIdentity.IsPresent
+            TenantId                                   = $TenantId
+            AccessTokens                               = $AccessTokens
+        }
+
+        Write-Verbose -Message 'Found Migration config '
+        Write-Verbose -Message "Get-TargetResource Result: `n $(Convert-M365DscHashtableToString -Hashtable $result)"
+        return $result
+    }
+    catch
+    {
+        New-M365DSCLogEntry -Message 'Error retrieving data:' `
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+
+        return $nullReturn
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        [ValidateSet('Yes')]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.String]
+        $Identity,
+
+        [Parameter()]
+        [System.Boolean]
+        $CanSubmitNewBatch,
+
+        [Parameter()]
+        [System.Boolean]
+        $SupportsCutover,
+
+        [Parameter()]
+        [System.Boolean]
+        $IsValid,
+
+        [Parameter()]
+        [System.String]
+        $ObjectState,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [System.String]
+        $CertificatePath,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $CertificatePassword,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace 'MSFT_', ''
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    Write-Verbose -Message 'Setting configuration of Migration config'
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'ExchangeOnline' `
+        -InboundParameters $PSBoundParameters
+
+    $MigrationConfigParams = [System.Collections.Hashtable]($PSBoundParameters)
+    $MigrationConfigParams.Remove('Ensure') | Out-Null
```

Comment:
For the authentication params, let's use the Remove-M365DSCAuthenticationParameter cmdlet instead. That way if we are to add additional auth params in the future, we'll only have a central place to remove the entry.